scaleway/resource_k8s_cluster.go

```diff
@@ -222,6 +222,34 @@ func resourceScalewayK8SCluster() *schema.Resource {
 				Description: "The status of the cluster",
 			},
 		},
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+			autoUpgradeEnable, okAutoUpgradeEnable := diff.GetOkExists("auto_upgrade.0.enable")
+			_, okAutoUpgradeStartHour := diff.GetOkExists("auto_upgrade.0.maintenance_window_start_hour")
+			_, okAutoUpgradeDay := diff.GetOk("auto_upgrade.0.maintenance_window_day")
+
+			if okAutoUpgradeEnable {
+				// check if either all or none of the auto upgrade attribute are set.
+				// if one auto upgrade attribute is set, they all must be set.
+				// if none is set, auto upgrade attributes will be computed.
+				if !(okAutoUpgradeDay && okAutoUpgradeStartHour) {
```

Comment:
Hum the comment is not aligned with the code, the behaviour changed in https://github.com/scaleway/terraform-provider-scaleway/pull/730 
Here if both the `okAutoUpgradeDay` and the `okAutoUpgradeStartHour` are false, an error is raised. Guess it should also be modified in the apply section. a simple `if okAutoUpgradeDay != okAutoUpgradeStartHour` instead of this one should be enough (if that's what is wanted)

Reply:
idem, code removed